File: code_corpus/manual/sort_6.m
function dataOut = shellSort(data)
lenD = size(data,2);
inc = cast(lenD/2,'int16');
while(inc>0)
  for i=inc:lenD
    tmp = data(i);
    j = i;
    while(j>inc && data(j-inc)>tmp)
      data(j) = data(j-inc);
      j = j-inc;
    end
    data(j) = tmp;
  end
  inc = cast(floor(double(inc) /2),'int16');
end
dataOut = data;

File: code_corpus/manual/sort_6.cpp
void SortAlgo::shellSort(int data[], int lenD)
{
  int inc = lenD/2;
  while(inc>0){
    for(int i=inc;i<lenD;i++){
      int tmp = data[i];
      int j = i;
      while(j>=inc && data[j-inc]>tmp){
        data[j] = data[j-inc];
        j = j-inc;
      }
      data[j] = tmp;
    }
  inc = (inc /2);
  }
}
